app/services/chargeable_zones_service.rb

```diff
@@ -32,14 +33,21 @@ def initialize(vehicle_details:)
   def call
     return [] if chargeable_zones_data.empty?
 
-    chargeable_vehicles_ids = non_dvla ? non_dvla_data : dvla_data
+    chargeable_vehicles_ids = select_chargeable_vehicles_ids
     chargeable_zones_data.filter_map do |zone|
       Caz.new(zone) if zone['cleanAirZoneId'].in?(chargeable_vehicles_ids)
     end.sort_by(&:name)
```

Comment:
```suggestion
    chargeable_zones_data.filter_map do |zone|
      Caz.new(zone) if zone['cleanAirZoneId'].in?(chargeable_vehicles_ids)
    end.sort_by(&:name)
```
and rename method to `chargeable_vehicles_ids`